packages/EasyApiPlatform/src/Filter/AdvancedSearchFilter.php

```diff
@@ -0,0 +1,485 @@
+                && is_array($this->properties[$possibleFilterParameter])
+            ) {
+                $filterParameter = $possibleFilterParameter;
+
+                $property = (string)array_keys($this->properties[$filterParameter])[0];
+                $strategy = array_values($this->properties[$filterParameter])[0];
+
+                $value = $value[$valueKey];
+            }
+        }
+
+        if (
+            $value === null ||
+            $this->isPropertyEnabled($filterParameter, $resourceClass) === false ||
+            $this->isPropertyMapped($property, $resourceClass, true) === false
+        ) {
+            return;
+        }
+
+        $alias = $queryBuilder->getRootAliases()[0];
+        $field = $property;
+
+        $values = $this->normalizeValues((array)$value, $property);
+        if ($values === null) {
+            return;
+        }
+
+        $associations = [];
+        if ($this->isPropertyNested($property, $resourceClass)) {
+            [$alias, $field, $associations] = $this->addJoinsForNestedProperty(
+                $property,
+                $alias,
+                $queryBuilder,
+                $queryNameGenerator,
+                $resourceClass,
+                Join::INNER_JOIN
+            );
+        }
+
+        $caseSensitive = true;
+        $strategy = $strategy ?? ($this->properties[$filterParameter] ?? self::STRATEGY_EXACT);
+
+        // prefixing the strategy with i makes it case-insensitive
+        if (str_starts_with($strategy, 'i')) {
+            $strategy = substr($strategy, 1);
+            $caseSensitive = false;
+        }
+
+        $metadata = $this->getNestedMetadata($resourceClass, $associations);
+
+        if ($metadata->hasField($field)) {
+            if ($field === 'id') {
+                $values = array_map([$this, 'getIdFromValue'], $values);
+            }
+
+            if ($this->hasValidValues($values, $this->getDoctrineFieldType($property, $resourceClass)) === false) {
+                $this->logger->notice('Invalid filter ignored', [
+                    'exception' => new InvalidArgumentException(sprintf(
+                        'Values for field "%s" are not valid according to the doctrine type.',
+                        $field
+                    )),
+                ]);
+
+                return;
+            }
+
+            $this->addWhereByStrategy(
+                $strategy,
+                $queryBuilder,
+                $queryNameGenerator,
+                $alias,
+                $field,
+                $values,
+                $caseSensitive
+            );
+
+            return;
+        }
+
+        // metadata doesn't have the field, nor an association on the field
+        if ($metadata->hasAssociation($field) === false) {
+            return;
+        }
+
+        $values = array_map([$this, 'getIdFromValue'], $values);
+
+        $associationResourceClass = (string)$metadata->getAssociationTargetClass($field);
```

Comment:
Why don't we need this check here?
```
if (null !== $this->identifiersExtractor) {
```

Reply:
Would you please recheck with `\ApiPlatform\Doctrine\Orm\Filter\SearchFilter`?

Reply:
I see, thank you.